From e3c8b59b7fd5036435284b6b15defb3afc4cb518 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Thu, 8 Mar 2018 11:49:31 +0800
Subject: [PATCH 398/767] Revert "drm/fsl-dcu: Fix DCU pixel clock issue in
 suspend/resume functions"

This reverts commit bd296cfffed03f25856ab77c28de6e8f306d0a01.

As upstream commit c8111b1885d3a802e57f895e77aaa1e88de57282
"drm/fsl-dcu: avoid disabling pixel clock twice on suspend" can also fix
the original issue, the commit bd296cfffed03f25856ab77c28de6e8f306d0a01
needs to be reverted.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c b/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
index 3447c1c0a4de..80232321a244 100644
--- a/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
+++ b/drivers/gpu/drm/fsl-dcu/fsl_dcu_drm_drv.c
@@ -221,12 +221,6 @@ static int fsl_dcu_drm_pm_resume(struct device *dev)
 		return ret;
 	}
 
-	ret = clk_prepare_enable(fsl_dev->pix_clk);
-	if (ret < 0) {
-		dev_err(dev, "failed to enable dcu pix clk\n");
-		return ret;
-	}
-
 	if (fsl_dev->tcon)
 		fsl_tcon_bypass_enable(fsl_dev->tcon);
 	fsl_dcu_drm_init_planes(fsl_dev->drm);
@@ -343,12 +337,6 @@ static int fsl_dcu_drm_probe(struct platform_device *pdev)
 		goto disable_clk;
 	}
 
-	ret = clk_prepare_enable(fsl_dev->pix_clk);
-	if (ret < 0) {
-		dev_err(dev, "failed to enable dcu pix clk\n");
-		return ret;
-	}
-
 	fsl_dev->tcon = fsl_tcon_init(dev);
 
 	drm = drm_dev_alloc(driver, dev);
-- 
2.17.0

